### PR TITLE
Fix issue with verify email redirect using Auth0

### PR DESCRIPTION
**Pull request by o0charlie0o**
When clicking the verify email link in an Auth0 email upon success Auth0 redirects to the main url defined with a bunch of search params. One of those params happens to be `code=` which `restoreAuthState` looks for in the `AuthProvider`. It does not however include the `state` param causing the rest of this function to fail with an invalid state error. This PR makes sure both `code` and `state` exist before continuing with the redirect callback.

Example of the url Auth0 redirects to:
`https://www.sample-website.com/?supportSignUp=true&supportForgotPassword=true&email=email%40sample-website.com&message=Your%20email%20was%20verified.%20You%20can%20continue%20using%20the%20application.&success=true&code=success#`

### CODE REVIEW

**Comment by peterp** [comment]
@o0charlie0o anything I can do to help this along?

**Comment by o0charlie0o** [comment]
@peterp I think it mainly needs to be tested although I have done that locally without issue. I put it in draft due to that failing test but I'm not really sure why it failed at the moment.

**Comment by peterp** [comment]
Let me re-run the tests, since those failures look completely unrelated.

File: packages/auth/src/authClients/auth0.ts [hunk 1]
```diff
@@ -14,7 +14,10 @@ export const auth0 = (client: Auth0): AuthClientAuth0 => {
     type: 'auth0',
     client,
     restoreAuthState: async () => {
-      if (window?.location?.search?.includes('code=')) {
+      if (
+        window?.location?.search?.includes('code=') &&
+        window?.location?.search?.includes('state=')
```
Comment (peterp) [inline packages/auth/src/authClients/auth0.ts:19]:
```suggestion
        global?.location?.search?.includes('code=') &&
        global?.location?.search?.includes('state=')
```

**Review by peterp — APPROVED** [review]

**Pull request merged**